lib/processors/libraryLessGenerator.js

```diff
@@ -0,0 +1,172 @@
+			// Skip processing when no imports are found
+			return fileContent;
+		}
+		const replacements = await Promise.all(imports.map(async (importMatch) => {
+			const baseDir = posixPath.dirname(filePath);
+			const resolvedFilePath = posixPath.resolve(baseDir, importMatch.path);
+			importMatch.content = await this.resolveLessImport(importMatch.path, resolvedFilePath, baseDir);
+			return importMatch;
+		}));
+
+		// Apply replacements in reverse order to not modify the relevant indices
+		const array = Array.from(fileContent);
+		for (let i = replacements.length - 1; i >= 0; i--) {
+			const replacement = replacements[i];
+			if (!replacement.content) {
+				continue;
+			}
+			array.splice(
+				/* index   */ replacement.matchStart,
+				/* count   */ replacement.matchLength,
+				/* insert  */ replacement.content
+			);
+		}
+		return array.join("");
+	}
+	async resolveLessImport(originalFilePath, resolvedFilePath, baseDir) {
+		// Rewrite base.less imports
+		const baseLessMatch = BASE_LESS_PATTERN.exec(resolvedFilePath);
+		if (baseLessMatch) {
+			let baseLessThemeName = baseLessMatch[1];
+			if (baseLessThemeName === "base") {
+				baseLessThemeName = "baseTheme";
+			}
+
+			const baseLessPath = LibraryLessGenerator.getPathToRoot(baseDir) +
+				"Base/baseLib/" + baseLessThemeName + "/base.less";
+			return "@import \"" + baseLessPath + "\"; /* ORIGINAL IMPORT PATH: \"" + originalFilePath + "\" */\n";
+		}
+
+		// Rewrite library imports to correct file name
+		if (posixPath.basename(resolvedFilePath) === "library.source.less") {
+			return `@import "${originalFilePath.replace(/library\.source\.less$/, "library.less")}";`;
+		}
+
+		// Excluding global.less within sap.ui.core
+		// It must be imported by the Theme Designer (also see declaration in sap/ui/core/.theming)
+		if (GLOBAL_LESS_PATTERN.test(resolvedFilePath)) {
+			return null;
+		}
+
+		/*
+		 * Throw error in case of files which are not in the same directory as the current file because
+		 * inlining them would break relative URLs.
+		 * Keeping the import is also not possible since only "library.less" and "global.less" are
+		 * configured to be available to the Theme Designer (.theming generated in generateThemeDesignerResources).
+		 *
+		 * A possible solution would be to rewrite relative URLs when inlining the content.
+		 */
+		const relativeFilePath = posixPath.relative(baseDir, resolvedFilePath);
+		if (relativeFilePath.includes(posixPath.sep)) {
+			throw new Error(
+				`libraryLessGenerator: Unsupported import of file '${resolvedFilePath}'. ` +
+				`Stylesheets must be located in the theme directory '${baseDir}' (no sub-directories)`
+			);
+		}
+
+		let importedFileContent;
+		try {
+			importedFileContent = await this.readFile(resolvedFilePath);
+		} catch (err) {
+			if (err.code === "ENOENT") {
+				throw new Error(
+					`libraryLessGenerator: Unable to resolve import '${originalFilePath}' from '${baseDir}'\n` +
+					err
+				);
+			} else {
+				throw err;
```

Comment:
Uncovered line (and branch). Should be easy to test, just like the other branch.